Added Dockerfile of maurorigo

maurorigo opened the pull request.

luca-heltai commented on ciao.md:
```diff
@@ -0,0 +1 @@
+Ciao
```
Could you remove this file?

luca-heltai commented on tests/rigo/ciao.m:
```diff
@@ -0,0 +1 @@
+Ciao
```
And also this?

maurorigo replied:
Oh I'm sorry, I forgot my pull request was still open, those files weren't supposed to end up in it.

luca-heltai commented on tests/rigo/test.py:
```diff
@@ -0,0 +1,14 @@
+
+key = jax.random.PRNGKey(13)
+arrjnp = jax.random.permutation(key, arr)
+print(f"Permuted with jax: {arrjnp}.")
```
Can you add an assert that checks some conditions? In this test, technically, you are not testing anything, just running jax. 

Make sure, for example, that the `arrjnp` has the right size (just to make sure your jax installation is not doing anything stupid).

maurorigo replied:
Sure! Thank you

maurorigo commented:
Note: this was my original pull request when we just had to push a working Dockerfile, it is not yet the one with github actions. For that one, I'm using a different branch, I've updated the image and I've added more tests. I may open the pull request within the weekend. The mess comes from the fact that I created this pull request from my main branch.

maurorigo commented:
I've now noticed that with the last commit I would be deleting the contents of tests/heltai (is this actually the case?). I wanted to try to fix this, so I fetched and pulled upstream main on my local main. The problem now is that my local main branch doesn't show the "tests" folder, but it keeps saying I'm up to date with upstream main. I've also tried to merge my main with the commit where the contents of tests/heltai were added, but it still tells me I'm up to date without showing the "tests" folder. What should I do?

luca-heltai commented:
Close this pull request, checkout a new branch from your main, fetch upstream, and rebase on upstream, i.e.,:

```
git checkout -b rigo-docker
git fetch upstream
git rebase upstream/master
```
if you have conflicts during this process, then abort the rebase
```
git rebase --abort
```
and do
```
git reset upstream/master
git add docker/maurorigo
git commit -m 'Added docker file.'
```
the reset is not a hard reset, so it will maintain your changes, and show the diffs between your branch and master. Drop any changes unrelated to `docker/maurorigo`. :)

maurorigo commented:
Alright, thank you!
